### PR TITLE
Fix Malformed Table

### LINKED ISSUE #520: Malformed table in DevelopmentCodeFragment doc
Hi,

The [Design Rules](https://phpmd.org/rules/design.html) doc has an error under `DevelopmentCodeFragment` table.

```
Malformed table. Text in column margin in table line 4.
```

Regards!

### PR DESCRIPTION
**Pull request by tariq86**
Fixes #520

**What**: The main changes in this PR are in the XSLT config file (`./src/site/scripts/pmd.xsl`) used to regenerate the RST ruleset pages. Most of the other changes are all from re-running the `update-rules.php` script after updating the config.

**How**: In order to fix document generation, I updated the config to use the longest widths for all columns. If the column length for "Default Value" has a minimum of 15.

**Why**: This should fix table formatting for all properties with long names, default values, or descriptions. As reported in #520, an example of this bug [can be seen here](https://phpmd.org/rules/design.html#developmentcodefragment).

---

In addition to fixing the doc. generator, this PR also includes one small change to restore builds on AppVeyor (update php7.1 references to php7.2), **and** fixes unit tests by updating PHPUnit.

### CODE REVIEW

**Comment by tariq86** [comment]
I've rebased and fixed the merge errors, but now the PHP5.x tests are failing on Travis CI...looks like the composer.lock file changes are preventing certain dependencies from being installed in 5.x environments.

**Comment by ravage84** [comment]
@tariq86 yes it seem so. They shouldn't fail, as the last commit to master went through:
https://travis-ci.org/phpmd/phpmd/builds/550368531

File: src/site/rst/rules/design.rst [hunk 1]
```diff
@@ -141,7 +149,7 @@ This rule has the following properties:
 =================================== =============== ============================================
  Name                                Default Value   Description                                
 =================================== =============== ============================================
- maximum                             13              Maximum number of acceptable dependencies.
+ maximum                             13              Maximum number of acceptable dependencies.
```
Comment (ravage84) [inline src/site/rst/rules/design.rst]: I think this trailing white space is unnecessary, please remove.

File: src/site/rst/rules/design.rst [hunk 2]
```diff
@@ -182,43 +191,46 @@ Since: PHPMD 2.7.0
...
 
   class Foo {
-
-      public function bar()
-      {
-          try {
-              // ...
-          } catch (Exception $e) {} // empty catch block
-      }
+
```
Comment (ravage84) [inline src/site/rst/rules/design.rst]: The indentation of the following lines are wrong, it seems. Can you fix it, please?

File: src/site/rst/rules/design.rst [hunk 3]
```diff
@@ -182,43 +191,46 @@ Since: PHPMD 2.7.0
...
   }
 
 CountInLoopExpression
 =====================
 
 Since: PHPMD 2.7.0
 
-Using count/sizeof in loops expressions is considered bad practice and is a potential source of
-many bugs, especially when the loop manipulates an array, as count happens on each iteration.
+Using count/sizeof in loops expressions is considered bad practice and is a potential source of many bugs, especially when the loop manipulates an array, as count happens on each iteration.
```
Comment (ravage84) [inline src/site/rst/rules/design.rst]: Why the change? The lines shouldn't be so long.

File: src/site/rst/rules/design.rst [hunk 4]
```diff
@@ -182,43 +191,46 @@ Since: PHPMD 2.7.0
...
+
 
 Example: ::
 
   class Foo {
-
+
```
Comment (ravage84) [inline src/site/rst/rules/design.rst]: Is this necessary? If not, please fix it.

File: src/site/rst/rules/design.rst [hunk 5]
```diff
@@ -182,43 +191,46 @@ Since: PHPMD 2.7.0
...
   class Foo {
-
+  
     public function bar()
     {
-      $arr = array();
-
-      for ($i = 0; count($arr); $i++) {
+      $array = array();
+
```
Comment (ravage84) [inline src/site/rst/rules/design.rst]: Is this necessary? I guess not.

File: src/site/rst/rules/index.rst [hunk 6]
```diff
@@ -4,81 +4,134 @@ Current Rulesets
 
 List of rulesets and rules contained in each ruleset.
 
-- `Clean Code Rules <#clean-code-rules>`_: The Clean Code ruleset contains rules that enforce a clean code base. This includes rules from SOLID and object calisthenics.
```
Comment (ravage84) [inline src/site/rst/rules/index.rst]: These lines were changed recently, please change the rule generation accordingly.

File: src/site/rst/rules/index.rst [hunk 7]
```diff
@@ -4,81 +4,134 @@ Current Rulesets
...
+__ index.html#code-size-rules
+__ index.html#controversial-rules
+__ index.html#design-rules
+__ index.html#naming-rules
+__ index.html#unused-code-rules
 
 Clean Code Rules
 ================
 
-- `BooleanArgumentFlag <cleancode.html#booleanargumentflag>`_: A boolean flag argument is a reliable indicator for a violation of the Single Responsibility Principle (SRP). You can fix this problem by extracting the logic in the boolean flag into its own class or method.
```
Comment (ravage84) [inline src/site/rst/rules/index.rst]: These lines were changed recently, please change the rule generation accordingly.

File: src/site/rst/rules/index.rst [hunk 8]
```diff
@@ -4,81 +4,134 @@ Current Rulesets
...
+__ cleancode.html#booleanargumentflag
+__ cleancode.html#elseexpression
+__ cleancode.html#ifstatementassignment
+__ cleancode.html#staticaccess
+__ cleancode.html#duplicatedarraykey
 
 Code Size Rules
 ===============
 
-- `CyclomaticComplexity <codesize.html#cyclomaticcomplexity>`_: Complexity is determined by the number of decision points in a method plus one for the method entry. The decision points are 'if', 'while', 'for', and 'case labels'. Generally, 1-4 is low complexity, 5-7 indicates moderate complexity, 8-10 is high complexity, and 11+ is very high complexity.
```
Comment (ravage84) [inline src/site/rst/rules/index.rst]: These lines were changed recently, please change the rule generation accordingly.

File: src/site/rst/rules/index.rst [hunk 9]
```diff
@@ -4,81 +4,134 @@ Current Rulesets
...
+__ codesize.html#excessivepubliccount
+__ codesize.html#toomanyfields
+__ codesize.html#toomanymethods
+__ codesize.html#toomanypublicmethods
+__ codesize.html#excessiveclasscomplexity
 
 Controversial Rules
 ===================
 
-- `Superglobals <controversial.html#superglobals>`_: Accessing a super-global variable directly is considered a bad practice. These variables should be encapsulated in objects that are provided by a framework, for instance.
```
Comment (ravage84) [inline src/site/rst/rules/index.rst]: These lines were changed recently, please change the rule generation accordingly.

File: src/site/rst/rules/index.rst [hunk 10]
```diff
@@ -4,81 +4,134 @@ Current Rulesets
...
+__ controversial.html#camelcaseclassname
+__ controversial.html#camelcasepropertyname
+__ controversial.html#camelcasemethodname
+__ controversial.html#camelcaseparametername
+__ controversial.html#camelcasevariablename
 
 Design Rules
 ============
 
-- `ExitExpression <design.html#exitexpression>`_: An exit-expression within regular code is untestable and therefore it should be avoided. Consider to move the exit-expression into some kind of startup script where an error/exception code is returned to the calling environment.
```
Comment (ravage84) [inline src/site/rst/rules/index.rst]: These lines were changed recently, please change the rule generation accordingly.

File: src/site/rst/rules/index.rst [hunk 11]
```diff
@@ -4,81 +4,134 @@ Current Rulesets
...
+__ design.html#depthofinheritance
+__ design.html#couplingbetweenobjects
+__ design.html#developmentcodefragment
+__ design.html#emptycatchblock
+__ design.html#countinloopexpression
 
 Naming Rules
 ============
 
-- `ShortVariable <naming.html#shortvariable>`_: Detects when a field, local, or parameter has a very short name.
```
Comment (ravage84) [inline src/site/rst/rules/index.rst]: These lines were changed recently, please change the rule generation accordingly.

File: src/site/rst/rules/index.rst [hunk 12]
```diff
@@ -4,81 +4,134 @@ Current Rulesets
...
+__ naming.html#longvariable
+__ naming.html#shortmethodname
+__ naming.html#constructorwithnameasenclosingclass
+__ naming.html#constantnamingconventions
+__ naming.html#booleangetmethodname
 
 Unused Code Rules
 =================
 
-- `UnusedPrivateField <unusedcode.html#unusedprivatefield>`_: Detects when a private field is declared and/or assigned a value, but not used.
```
Comment (ravage84) [inline src/site/rst/rules/index.rst]: These lines were changed recently, please change the rule generation accordingly.

**Comment by ravage84** [comment]
> In addition to fixing the doc. generator, this PR also includes one small change to restore builds on AppVeyor (update php7.1 references to php7.2), and fixes unit tests by updating PHPUnit.

Is it possible to extract this change into a separate PR? I'd really appreaciate it.

**Comment by tariq86** [comment]
Yeah sure! I think it would be easier to start with two fresh branches and cherry-pick the relevant commits. I’ll do that now and open the separate PRs...I’ll comment here once that’s done so we can close this one

**Comment by kylekatarnls** [comment]
Hi @tariq86, we changed tables for pretty-style ones:
```
+-----------------------------------+---------------+------------------------------------------------------------+
| Name                              | Default Value | Description                                                |
+===================================+===============+============================================================+
| exceptions                        |               | Comma-separated class name list of exceptions              |
+-----------------------------------+---------------+------------------------------------------------------------+
```

So the malformed tables part of your PR is no longer relevant.

As you propose, if you could rather create 1 PR for each fix, each fix would have more chances to get merged as is. The problem of PRs handling multiple issues is while only 1 change request left, ready changes cannot be merged.
